pkg/vz/save_restore_arm64.go

```diff
@@ -0,0 +1,76 @@
+//go:build darwin && arm64 && !no_vz
+
+package vz
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/Code-Hex/vz/v3"
+	"github.com/sirupsen/logrus"
+)
+
+func saveVM(vm *vz.VirtualMachine, machineStatePath string) error {
+	if !vm.CanPause() {
+		return fmt.Errorf("can't pause the VZ machine")
+	}
+
+	// Remove the old machine state file if it exists,
+	// because saving the machine state will fail if the file already exists.
+	if err := os.Remove(machineStatePath); err != nil && !errors.Is(err, os.ErrNotExist) {
+		logrus.WithError(err).Errorf("Failed to remove the old VZ machine state file %q", machineStatePath)
+		return err
+	}
```

Comment:
We can write machine state to `vz-machine-state.tmp` and then replace the current state by renaming `vz-machine-state.tmp` to `vz-machine-state`. This ensures that we always have consistent state, either the previous one or the new one.

Reply:
Currently, I don't see the value in keeping the `vz-machine-state` after a resume. This is because the current implementation does not save disk snapshots, and there is no mechanism to revert the guest OS's changes to the diffdisk after a resume to restore the state to match when the `vz-machine-state` was saved. 

By the time we reach this point, the guest OS should already have gone through `StatusRunning`, so I don't think it's worth complicating things to maintain consistency with the previous `vz-machine-state`.

Reply:
Makes sense. Maybe add a comment here to explain why we don't keep the state?